patch_rebaser/patch_rebaser.py

```diff
@@ -99,12 +100,54 @@ def get_rebaser_config():
             "Configuration file {0} not found.".format(config_file)
         )
 
-    rebaser_config = configparser.ConfigParser()
-    rebaser_config.read(config_file)
-    return rebaser_config
+    parser = configparser.ConfigParser(defaults)
+    parser.read(config_file)
+    return parser
+
+
+def get_rebaser_config(defaults=None):
+    """Return a tuple with the configuration information.
+
+    If the configuration is missing an option, the value from the
+    defaults dictionary is used. If the defaults doesn't contain the
+    value either, a configparser.NoOptionError exception is raised.
+
+       :param dict defaults: Default values for config values
+
+    """
+    default_options = ['dev_mode', 'remote_name', 'git_name', 'git_email',
+                       'packages_to_process', 'dlrn_projects_ini']
+    distroinfo_options = ['patches_repo_key']
+
+    RebaserConfig = namedtuple('RebaserConfig',
+                               default_options + distroinfo_options)
+
+    parser = get_rebaser_configparser(defaults)
+
+    options = {}
+    for opt in default_options:
+        if opt == 'dev_mode':
+            options[opt] = parser.getboolean('DEFAULT', opt)
+        elif opt == 'packages_to_process':
+            pkgs = parser.get('DEFAULT', opt)
+            if pkgs:
+                pkgs = pkgs.split(",") if "," in pkgs else [pkgs]
+            options[opt] = pkgs
+        else:
+            options[opt] = parser.get('DEFAULT', opt)
+
+    if not parser.has_section('distroinfo'):
+        parser.add_section('distroinfo')
+    options['patches_repo_key'] = parser.get('distroinfo', 'patches_repo_key')
```

Comment:
Maybe I am misunderstanding, but is patches_repo_key going to be added twice to the options?  It looks like you add it above, on lines 122-123, but not in the distroinfo section.

Reply:
Line 122/123 only creates the namedtuple and doesn't assign any value. Basically this all flattens the config and ignores the sections (every option lives at the same level), which is fine at this point as there are no option name conflicts. The 'defaults' dict for ConfigParser works the same way.

I sort of regret introducing a separate section for distroinfo at this point, since it seems to add complexity for little benefit, though that's done.

Does this help or did I misunderstand the question??

Reply:
Perfect, thanks, I get it now.  Thanks for clarifying!